### PR TITLE
Fixing lighting issues with UsdPreviewSurface

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
We recently debugged a scene that failed to render correctly with the V2
Light API. The main issue with shadows was fixed in another PR, but there
were still 3 other problems where the Light API V1 was not on-par with USD
and the V2 Light API.

1- Directional light affects IBL lighting

Create a test scene:

- 20x20 plane at the origin, usdPreviewSurface shading, 50% metalness
- aiSkyDomeLight with an environment map
- Directional light with *zero* intensity

Rotate the directional light, and watch how it affects the lighting of the plane. That light has zero intensity, so it should have no effect.

2- textured occlusion does not affect environment lighting

One of the assets in the scene had a textured occlusion map which
correctly worked with scene lights, but not with aiSkyDome lights.

3- specular highlights from regular Maya lights did not have enough base
color for metallic workflows.

That one was fixed for aiSkyDome lights in PR #1356 but direct lighting
still used the old formula.

Fixes internal bugs MAYA-113254 MAYA-113272 and MAYA-113269

### CODE REVIEW

**Comment by JGamache-autodesk** [comment]
@dgovil , @ysiewappl Can you review and confirm you are OK with these fixes. Thanks!

File: lib/mayaUsd/render/vp2ShaderFragments/UsdPreviewSurfaceLightAPI1.xml [hunk 1]
```diff
@@ -65,7 +66,7 @@ limitations under the License.
         <connect from="mayaShaderGeom_Float4GetY.output" to="usdPreviewSurfaceLightingAPI1.NdotV" name="NdotV"/>
         <connect from="mayaShaderGeom_Float4GetZ.output" to="usdPreviewSurfaceLightingAPI1.NdotH" name="NdotH"/>
         <connect from="mayaShaderGeom_Float4GetW.output" to="usdPreviewSurfaceLightingAPI1.VdotH" name="VdotH"/>
-        <connect from="mayaShaderGeom_Float4GetW.output" to="usdPreviewSurfaceCombiner.VdotH" name="VdotH"/>
+        <connect from="mayaShaderGeom_Float4GetY.output" to="usdPreviewSurfaceCombiner.VdotH" name="NdotV"/>>
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2ShaderFragments/UsdPreviewSurfaceLightAPI1.xml:69]: Fixes issue 1. There is no H vector for environment maps, so Maya was re-using the last H it saw. Use VdotN instead.

File: lib/mayaUsd/render/vp2ShaderFragments/usdPreviewSurfaceLightingAPI1.xml [hunk 2]
```diff
@@ -185,7 +185,7 @@ usdPreviewSurfaceLightingAPI1(
         if (!useSpecularWorkflow) {
             float R = (1.0 - ior) / (1.0 + ior);
             float3 specColor = mix(float3(1.0, 1.0, 1.0), diffuseColor, metallic);
-            F0  = R * R * specColor;
+            F0 = mix(R * R * specColor, specColor, metallic);
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2ShaderFragments/usdPreviewSurfaceLightingAPI1.xml:188]: Fixes issue 3 by using the formula from [USD](https://github.com/PixarAnimationStudios/USD/commit/f11ab360)

File: lib/mayaUsd/render/vp2ShaderFragments/UsdPreviewSurfaceLightAPI1.xml [hunk 3]
```diff
@@ -86,6 +87,8 @@ limitations under the License.
         <connect from="metallicPassThrough.floatPassThrough" to="usdPreviewSurfaceCombiner.metallic" name="metallic"/>
         <connect from="iorPassThrough.floatPassThrough" to="usdPreviewSurfaceLightingAPI1.ior" name="ior"/>
         <connect from="iorPassThrough.floatPassThrough" to="usdPreviewSurfaceCombiner.ior" name="ior"/>
+        <connect from="occlusionPassThrough.floatPassThrough" to="usdPreviewSurfaceLightingAPI1.occlusion" name="occlusion"/>
+        <connect from="occlusionPassThrough.floatPassThrough" to="usdPreviewSurfaceCombiner.occlusion" name="occlusion"/>
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2ShaderFragments/UsdPreviewSurfaceLightAPI1.xml:91]: Fixes issue 2 by passing occlusion to the light combiner.

File: lib/mayaUsd/render/vp2ShaderFragments/UsdPreviewSurfaceLightAPI1.xml (hunk 1 above)
Comment (JGamache-autodesk) [reply lib/mayaUsd/render/vp2ShaderFragments/UsdPreviewSurfaceLightAPI1.xml:69]: Yes, indeed a typo in the comment.

**Comment by dgovil** [comment]
These changes look good to me! Thanks for making the update :-)

**Pull request merged**